cudamat/cudamat.py

```diff
@@ -235,6 +239,9 @@ def __init__(self, array, copy_to_device=True, copy_on_host=True):
                 if err_code:
                     raise generate_exception(err_code)
 
+            if not copy_on_host:
+                del self.numpy_array
+
```

Comment:
This is not what `copy_on_host` means -- it's fully covered by the `copy=copy_on_host` flag in `reformat` above. It is just a way to avoid an unneeded extra copy on the host, not a way to avoid linking the matrix to a numpy array at all. The extra copy is still enabled by default to not break backwards compatibility.